.gitlab-ci.yml

```diff
@@ -148,13 +162,25 @@ publish:tests:
     - export CI_BRANCH=${CI_COMMIT_BRANCH}
     - export CI_PR_NUMBER=${CI_COMMIT_BRANCH#pr_}
   script:
+    - "echo 'mode: set' > coverage.txt"
+    - for PLATFORM in linux mac; do
+    -   goveralls
+          -repotoken ${COVERALLS_TOKEN}
+          -service gitlab-ci
+          -jobid $CI_PIPELINE_ID
+          -covermode set
+          -flagname unittests:$PLATFORM
+          -coverprofile coverage-$PLATFORM.txt
+    -   grep -v '^mode:' coverage-$PLATFORM.txt >> coverage.txt
+    - done
+    # Combine all coverage reports.
     - goveralls
-      -repotoken ${COVERALLS_TOKEN}
-      -service gitlab-ci
-      -jobid $CI_PIPELINE_ID
-      -covermode set
-      -flagname unittests
-      -coverprofile coverage.txt
+        -repotoken ${COVERALLS_TOKEN}
+        -service gitlab-ci
+        -jobid $CI_PIPELINE_ID
+        -covermode set
+        -flagname unittests
+        -coverprofile coverage.txt
 
```

Comment:
This is a bit of a hack: we are reporting three independent coverage reports (linux, mac and merge).

The result should be the same but the correct way to do this is using `parallel` feature of coveralls: the first to calls use `-parallel` flag and then we have a curl call like [this one](https://github.com/mendersoftware/mendertesting/blob/master/.gitlab-ci-check-golang-unittests.yml#L117).

Reply:
That's better! I have changed it.